File: src/main/java/com/pedigree/render/GraphicsContext.java
package com.pedigree.render;

public interface GraphicsContext {
    void setStrokeColor(int r, int g, int b, double a);
    void setFillColor(int r, int g, int b, double a);
    void setLineWidth(double w);
    void setFontSize(double size);
    void drawLine(double x1, double y1, double x2, double y2);
    void fillRect(double x, double y, double w, double h);
    void drawRect(double x, double y, double w, double h);
    void drawText(String text, double x, double y);
}

File: src/main/kotlin/com/family/tree/render/GraphicsContext.kt
package com.family.tree.render

interface GraphicsContext {
    fun setStrokeColor(r: Int, g: Int, b: Int, a: Double)
    fun setFillColor(r: Int, g: Int, b: Int, a: Double)
    fun setLineWidth(w: Double)
    fun setFontSize(size: Double)
    fun drawLine(x1: Double, y1: Double, x2: Double, y2: Double)
    fun fillRect(x: Double, y: Double, w: Double, h: Double)
    fun drawRect(x: Double, y: Double, w: Double, h: Double)
    fun drawText(text: String, x: Double, y: Double)
    fun measureTextWidth(text: String): Double
}
